airflow/utils/timezone.py

```diff
@@ -74,6 +77,18 @@ def utc_epoch() -> dt.datetime:
     return result
 
 
+def in_timezone(datetime: DateTime, tz: Timezone, dst_rule: str = pendulum.PRE_TRANSITION) -> DateTime:
```

Comment:
pendulum literals PRE_TRANSITION, PRE_TRANSITION will be removed into the pendulum 3 in favor of of [`datetime.datetime.fold`](https://docs.python.org/3/library/datetime.html#datetime.datetime.fold) maybe we could use fold here, and use `dst_rule` depends on fold value `0` or `1`

Seems like PRE_TRANSITION use same behaviour as fold=0 (default in datetime.datetime) and  POST_TRANSITION same as fold=1, but better to check in [pendulum 2.1.2 code](https://github.com/sdispater/pendulum/blob/2.1.2/pendulum/tz/timezone.py#L92)

Reply:
Yeah they're the same, I can look to change this, the necessary functionality involves switching between pre and post based on which direction you're moving and which DST transition it is, to account for doing the extra hour both pre and post transition

Reply:
Yeah my point that in some point in the future (I hope near) pendulum 3 released and we need to migrate and most I guess we need to support two major versions. And if it possible to use `fold` as a parameter to functions now then transition would be less painful then. ❤️

Reply:
Pendulum 2.1.2 actually already supports using `fold` to control transition.

https://github.com/sdispater/pendulum/blob/d9aa101fe76852b75cba14d686ede923641c5b4d/pendulum/tz/timezone.py#L100-L103

This was added 6 years ago. I didn’t check which version. It should be safe to use it.

https://github.com/sdispater/pendulum/commit/5be0094a89d629af5fd2d87164d1c0b8e2c8bde8

Reply:
@uranusjr thanks for looking into this, i will make the switch